support/cas-server-support-spnego/src/main/java/org/apereo/cas/config/SpnegoConfiguration.java

```diff
@@ -65,40 +64,57 @@ public BeanContainer<Authentication> spnegoAuthentications(
         JcifsConfig.SystemSettings.setUseSubjectCredsOnly(spnegoSystem.isUseSubjectCredsOnly());
 
         val props = casProperties.getAuthn().getSpnego().getProperties();
-        return BeanContainer.of(props.stream()
-            .map(p -> {
-                val c = new JcifsConfig();
-                val jcifsSettings = c.getJcifsSettings();
-                jcifsSettings.setJcifsDomain(p.getJcifsDomain());
-                jcifsSettings.setJcifsDomainController(p.getJcifsDomainController());
-                jcifsSettings.setJcifsNetbiosCachePolicy(p.getCachePolicy());
-                jcifsSettings.setJcifsNetbiosWins(p.getJcifsNetbiosWins());
-                jcifsSettings.setJcifsPassword(p.getJcifsPassword());
-                jcifsSettings.setJcifsServicePassword(p.getJcifsServicePassword());
-                jcifsSettings.setJcifsServicePrincipal(p.getJcifsServicePrincipal());
-                jcifsSettings.setJcifsSocketTimeout(Beans.newDuration(p.getTimeout()).toMillis());
-                jcifsSettings.setJcifsUsername(p.getJcifsUsername());
-                return new Authentication(jcifsSettings.getProperties());
-            })
-            .collect(Collectors.toList()));
+        return props.stream()
+                .map(p -> {
+                    val c = new JcifsConfig();
+                    val jcifsSettings = c.getJcifsSettings();
+                    jcifsSettings.setJcifsDomain(p.getJcifsDomain());
+                    jcifsSettings.setJcifsDomainController(p.getJcifsDomainController());
+                    jcifsSettings.setJcifsNetbiosCachePolicy(p.getCachePolicy());
+                    jcifsSettings.setJcifsNetbiosWins(p.getJcifsNetbiosWins());
+                    jcifsSettings.setJcifsPassword(p.getJcifsPassword());
+                    jcifsSettings.setJcifsServicePassword(p.getJcifsServicePassword());
+                    jcifsSettings.setJcifsServicePrincipal(p.getJcifsServicePrincipal());
+                    jcifsSettings.setJcifsSocketTimeout(Beans.newDuration(p.getTimeout()).toMillis());
+                    jcifsSettings.setJcifsUsername(p.getJcifsUsername());
+                    return new Authentication(jcifsSettings.getProperties());
+                })
+                .collect(Collectors.toList());
+    }
+
+    @RefreshScope(proxyMode = ScopedProxyMode.DEFAULT)
+    @Bean
+    @ConditionalOnMissingBean(name = "spnegoAuthenticationsPool")
+    public BlockingQueue<List<Authentication>> spnegoAuthenticationsPool(
+            final CasConfigurationProperties casProperties,
+            final ConfigurableApplicationContext applicationContext) {
+        val spnegoProperties = casProperties.getAuthn().getSpnego();
+        val poolSize = spnegoProperties.getPoolSize();
+        val spnegoAuthenticationPool = new ArrayBlockingQueue<List<Authentication>>(poolSize);
+        for (int i = 0; i < poolSize; i++) {
```

Comment:
You should be using streams here, and collecting the results into a list.

Reply:
You mean:

```java
val spnegoAuthenticationPool = new ArrayBlockingQueue<List<Authentication>>(poolSize);
IntStream.range(0, poolSize).forEach(i -> spnegoAuthenticationPool.add(buildSpnegoAuthentications(casProperties, applicationContext).toList()));
return spnegoAuthenticationPool;
```

?

Reply:
Yes but not exactly. 

Something closer to:

```
IntStream.range(0, poolSize)
	.map(entry -> buildSpnegoAuthentications(casProperties, applicationContext))
	.map(entry -> entry.toList())
   .flatmap(List::stream)
   .collect(toList());
```

Then the final result can be just added to the parent list in one call.

Reply:
Ah, OK. Up to you; If you find that it makes it hard to read, I am fine with keeping it as is.

Reply:
OK. Will keep it as is.